### PR TITLE
Extract to another class

### PR DESCRIPTION
**Pull request by TheZeta**
Please refer to the commit message. GUI redesigning was not done. It will be handled later. Some inconsistencies were noted as the algorithm did not yield the optimal result and sometimes not placing a single sensor at all. Trying to find the cause to this was difficult for me so a peer review is preferable. I restored the changes I made to see if the aforementioned errors were caused by my modifications, but I had the same problems.

### CODE REVIEW

File: src/main/java/com/targus/ui/WSNPrototype.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("JSON Files", "*.json"));
+            File f = fc.showSaveDialog(null);
+
+            if (f == null) {
+                return;
+            }
+
+            String src = f.getAbsolutePath();
+
+            BufferedWriter writer = Files.newBufferedWriter(Paths.get(src));
```
Comment (dogukanteber) [inline src/main/java/com/targus/ui/WSNPrototype.java:101]: We can extract this writing to JSON file operations into a separate private method and call it.

```java
private Map<String, Object> <method_name> () {}
```

File: src/main/java/com/targus/ui/WSNPrototype.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        }
+
+        String src = f.getAbsolutePath();
+
+        Point2D dimensions;
+        Point2D[] targetArray;
+        Point2D[] potentialPositionArray;
+
+        try {
+            Reader reader = Files.newBufferedReader(Paths.get(src));
```
Comment (dogukanteber) [inline src/main/java/com/targus/ui/WSNPrototype.java:156]: The same thing mentioned for JSON write can also be done here. We can write a private method that reads the data

File: src/main/java/com/targus/ui/WSNPrototype.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class WSNPrototype {
+    private int paneWidth;
+    private int paneHeight;
+    private final ArrayList<Point2D> potentialPositions = new ArrayList<Point2D>();
+
+    public ArrayList<Point2D> getPotentialPositions() {
```
Comment (dogukanteber) [inline src/main/java/com/targus/ui/WSNPrototype.java:26]: The class has a default constructor and initially, all class members will contain `null` or empty values.  Can we add another return value or exception for `null` and empty values?

File: src/main/java/com/targus/ui/WSNPrototype.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        return mProperty;
+    }
+
+    private final IntegerProperty kProperty = new SimpleIntegerProperty(1);
+
+    public IntegerProperty getKProperty() {
+        return kProperty;
+    }
+
+    private final DoubleProperty communicationRangeProperty = new SimpleDoubleProperty(100);
```
Comment (dogukanteber) [inline src/main/java/com/targus/ui/WSNPrototype.java:48]: In the `Constants` class, we can define `static final` variables for the magic numbers. After we define this variable, we can replace 100 with `Constants.DEFAULT_COMMUNICATION_RANGE`.

Also, we cannot alter the value of the communication range, at least I could not see a setter method. In the previous implementation, we could change this value. We need to define a setter method for this member and for all the other members

File: src/main/java/com/targus/ui/WSNPrototype.java [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    private int paneHeight;
+    private final ArrayList<Point2D> potentialPositions = new ArrayList<Point2D>();
+
+    public ArrayList<Point2D> getPotentialPositions() {
+        return potentialPositions;
+    }
+
+    private final ArrayList<Point2D> targets = new ArrayList<Point2D>();
+
+    public ArrayList<Point2D> getTargets() {
```
Comment (dogukanteber) [inline src/main/java/com/targus/ui/WSNPrototype.java:32]: There is no method for adding a single value or multiple values to `targets` and `potentialPositions` lists. We need to define two methods for `targets` and `potentialPositions` lists:

1) For adding a single element to the list
2) For adding multiple elements to the list

**Comment by dogukanteber** [comment]
I think we should not merge this unless the GUI is completely refactored. After the necessary changes are made, you can work on this branch to completely refactor the GUI. In my opinion, we should merge all the changes that are done regarding the GUI in one pull request.

File: src/main/java/com/targus/ui/WSNPrototype.java (hunk 1 above)
Comment (TheZeta) [reply src/main/java/com/targus/ui/WSNPrototype.java:101]: Adds to the complexity without a significant gain. So I thought it's better to keep it as is.

File: src/main/java/com/targus/ui/WSNPrototype.java (hunk 2 above)
Comment (TheZeta) [reply src/main/java/com/targus/ui/WSNPrototype.java:156]: Adds to the complexity without a significant gain. So I thought it's better to keep it as is.

File: src/main/java/com/targus/ui/WSNPrototype.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+            throw new RuntimeException(e);
+        }
+
+        paneWidth = (int) dimensions.getX();
+        paneHeight = (int) dimensions.getY();
+
+        Collections.addAll(targets, targetArray);
+        Collections.addAll(potentialPositions, potentialPositionArray);
+    }
+}
```
[the remark this answers is not in the document]
Comment (TheZeta) [reply src/main/java/com/targus/ui/WSNPrototype.java:196]: This review feels a bit phony.

**Comment by TheZeta** [comment]
For a long time, we held off on this pull request for the sole purpose of preserving the reviews made here which were used as guidelines for the new pull request focusing on implementing WSNPrototype. Now this pull request is not needed so it is abandoned.

**Pull request closed without merging**